pyrevolve/storage.py

```diff
@@ -1,10 +1,95 @@
 import numpy as np
+import datetime
 from functools import reduce
 from operator import mul
 
 from .logger import logger
 from .compression import CompressedObject
 import pickle
+import os
+
+
+class DiskStorage(object):
+    """Stores all checkpoints on a .dat binary file. The
+    []-operator is overloaded to return a file object with its
+    file-pointer placed according to the checkpoint key.
+    Revolve will typically use this as LIFO, but the storage also supports
+    random access. By default, the .dat file is removed when the object
+    is destroyed.
+    """
+
+    """Creates a .dat bynary file for storing checkpoints in
+    disk. Requires number of checkpoints andsize of one checkpoint.
+    """
+    def __init__(self, size_ckp, n_ckp, dtype, profiler,
+                 filename=None, remove_file=True):
+        self.size_ckp = size_ckp
+        self.n_ckp = n_ckp
+        self.dtype = dtype
+        self.profiler = profiler
+        if filename is None:
+            self.filename = "CKP_{}.dat".format(
+                datetime.datetime.now().strftime("%Y%m%d-%H%M%S"))
```

Comment:
We should be doing something better than this to generate filenames. Think shared filesystem and MPI. We wouldn't want multiple ranks clashing? Minimum would be to include PID in the filename. Ideally we want to generate names in a better way.

Reply:
You're. I have added the thread Id to the filename.

Reply:
It should be the process ID. Threads are sort of irrelevant to MPI.